### PR TITLE
Add generated files to gitignore

### PR DESCRIPTION
**Pull request by samuelmeuli**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | No
| Patch: Bug Fix?          | Yes
| Minor: New Feature?      | No
| Major: Breaking Change?  | No
| Tests Added + Pass?      | No
| Documentation Provided   | Yes (code comments)
| Any Dependency Changes?  | No
| License                  | Apache License, Version 2.0

Git will now ignore all `target` directories in apps generated by the archetype, as well as the ClientLibs generated using `aem-clientlib-generator`.

### CODE REVIEW

File: src/main/resources/archetype-resources/react-app/.gitignore [hunk 1]
```diff
@@ -28,4 +27,4 @@ out/
 
 # code coverage
 coverage/
-/coverage
\ No newline at end of file
+/coverage
```
Comment (grubyak) [inline src/main/resources/archetype-resources/react-app/.gitignore]: please remove duplicated `/coverage`

File: .gitignore [hunk 2]
```diff
@@ -16,7 +16,6 @@
 
 # Build/Maven directory
 target/
-target/
```
Comment (habansal) [inline .gitignore]: Just to be clear, can you clarify why we are deleting this `target` entry from here ?

Comment (samuelmeuli) [reply .gitignore]: It's a duplicate line

Comment (habansal) [reply .gitignore]: oops! :D

**Pull request merged**